lisa/node.py

```diff
@@ -424,9 +444,9 @@ def set_connection_info_by_runbook(
     def set_connection_info(
         self,
         address: str = "",
-        port: Optional[int] = 22,
+        port: int = 22,
         public_address: str = "",
-        public_port: Optional[int] = 22,
+        public_port: Optional[int] = None,
```

Comment:
Why change the default values?

Reply:
The issue here is that when we need to specify both `public_port` and `port`. In most cases they are same and specifying only one leads to default value of 22 for the other. For example, the following code will fail:

```
# setup connection to nested vm
    connection_info = ConnectionInfo(
        address=host.public_address,
        port=guest_port,
        username=guest_username,
        password=guest_password,
    )

    nested_vm = RemoteNode(Node(name=name), 0, name)
    nested_vm.set_connection_info(
        **fields_to_dict(connection_info, ["address", "port", "username", "password"])
    )
```

Reply:
In this case, it should specify both port and public_port, if it's not default or inconsistent. With change L449 and L465, the public port will be set to 22 as the guest_port together.